### PR TITLE
Introduce feature config API

### PR DESCRIPTION
**Pull request by smatting**
This PR introduces an API at `GET /feature-configs*` endpoints. See `docs/developer/features.md` for details.

## Checklist

 - [x] The **PR Title** explains the impact of the change.
 - [x] The **PR description** provides context as to why the change should occur and what the code contributes to that effect. This could also be a link to a JIRA ticket or a Github issue, if there is one.
 - [x] If end-points have been added or changed: the **endpoint / config-flag checklist** (see Wire-employee only backend [wiki page](https://github.com/zinfra/backend-wiki/wiki/Checklists)) has been followed.
 - [x] If a schema migration has been added, I ran **`make git-add-cassandra-schema`** to update the cassandra schema documentation.
 - [x] Section *Unreleased* of **CHANGELOG.md** contains the following bits of information:
   - [x] A line with the title and number of the PR in one or more suitable sub-sections.
   - [x] If /a: measures to be taken by instance operators.
   - [x] If /a: list of cassandra migrations.
   - [x] If public end-points have been changed or added: does nginz need upgrade?
   - [x] If internal end-points have been added or changed: which services have to be deployed in a specific order?

### CODE REVIEW

File: libs/wire-api/src/Wire/API/Routes/Public/Galley.hs [hunk 1]
```diff
@@ -353,8 +354,9 @@ data Api routes = Api
 
 type ServantAPI = ToServantApi Api
 
+-- | Deprecated. Use FeatureConfigGet.
```
Comment (fisx) [inline libs/wire-api/src/Wire/API/Routes/Public/Galley.hs:363]: it's not deprecated.  it's still useful for team admins who want to distinguish the settings on a user from the overridden settings on the team.

Comment (smatting) [reply libs/wire-api/src/Wire/API/Routes/Public/Galley.hs]: Removed it and updated the documenation.

**Review by fisx — CHANGES_REQUESTED** [review]
- Extend check list at `TeamFeatureName`.
- Add integration tests for non-team users.  (not sure about this one, have i missed any?  will they still pass?)
- There are TODOs again.

File: libs/wire-api/src/Wire/API/Team/Feature.hs [hunk 2]
```diff
@@ -156,6 +159,8 @@ instance FromByteString TeamFeatureName where
         Right "classifiedDomains" -> pure TeamFeatureClassifiedDomains
         Right t -> fail $ "Invalid TeamFeatureName: " <> T.unpack t
 
+-- TODO: how do we make this consistent with 'KnownTeamFeatureNameSymbol'?  add a test for
+-- that?  anyway do we really need both?
```
Comment (smatting) [inline libs/wire-api/src/Wire/API/Team/Feature.hs:163]: We need `KnownTeamFeatureNameSymbol` to generate servant routes

File: libs/wire-api/src/Wire/API/Routes/Public/Galley.hs [hunk 3]
```diff
@@ -372,6 +399,19 @@ type FeatureStatusDeprecatedPut featureName =
     :> ReqBody '[Servant.JSON] (TeamFeatureStatus featureName)
     :> Put '[Servant.JSON] (TeamFeatureStatus featureName)
 
+type FeatureConfigGet featureName =
+  Summary (AppendSymbol "Get feature config for feature " (KnownTeamFeatureNameSymbol featureName))
+    :> ZUser
+    :> "feature-configs"
```
[the remark this answers is not in the document]
Comment (fisx) [reply libs/wire-api/src/Wire/API/Routes/Public/Galley.hs:405]: never mind, changed my mind.

**Review by fisx — APPROVED** [review]

**Pull request merged**